embeddings/data/dataset.py

```diff
@@ -23,14 +21,7 @@ def __repr__(self) -> str:
         return type(self).__name__
 
 
-class LocalDataset(Dataset[T_path]):
-    def __init__(self, dataset: T_path, **load_dataset_kwargs: Any):
-        super().__init__()
-        self.dataset = dataset
-        self.load_dataset_kwargs = load_dataset_kwargs
-
-
-class HuggingFaceDataset(Dataset[str]):
+class LoadableDataset(Dataset[str]):
```

Comment:
I'd consider renaming `LoadableDataset` to some other name since probably all of our datasets will be loadable (is there any use of unloadable dataset?)

Reply:
I agree with that. I named it based on variables. Any suggestion how to improve it?

Reply:
maybe rename this one to Dataset and Dataset to BaseDataset?

Reply:
consider Base Dataset and Expandable / Extensible Dataset? Base is fine but just Dataset seems a little to general

Reply:
> consider Base Dataset and Expandable / Extensible Dataset? Base is fine but just Dataset seems a little to general

But in this case it is General as wall as name. Currently we don't need other object

Reply:
I renamed it to `BaseDataset` and `Dataset` accordingly, we can change this concept later in the general refactor. But in my view until we add another `Dataset` class I think this naming is ok.